### PR TITLE
Use pre-distributed MANIFEST files when available

### PR DESCRIPTION
**Pull request by cperciva**
As discussed over email.

### CODE REVIEW

File: src/share/poudriere/jail.sh [hunk 1]
```diff
@@ -591,15 +591,28 @@ install_from_ftp() {
...
-		fetch_file ${JAILMNT}/fromftp/MANIFEST ${URL}/MANIFEST
 		[ -s "${JAILMNT}/fromftp/MANIFEST" ] || err 1 "Empty MANIFEST file."
 		for dist in ${DISTS}; do
 			grep -q ${dist} ${JAILMNT}/fromftp/MANIFEST || continue
 			msg "Fetching ${dist} for FreeBSD ${V} ${ARCH}"
 			fetch_file ${JAILMNT}/fromftp/${dist}.txz ${URL}/${dist}.txz
+			MHASH=`awk "/^${dist}/ { print \\$2 }" ${JAILMNT}/fromftp/MANIFEST`
+			FHASH=`sha256 -q ${JAILMNT}/fromftp/${dist}`
+			if [ ${MHASH} != ${FHASH} ]; then
+				err 1 "${dist} checksum mismatch"
```
Comment (bdrewery) [inline src/share/poudriere/jail.sh]: Sigh. I can't believe this wasn't here already!

Comment (cperciva) [inline src/share/poudriere/jail.sh]: I admit that I was surprised!  It doesn't exist in the pre-9.0 downloading code either, but I didn't think it was worth updating code which only applies to unsupported releases.

File: src/share/poudriere/jail.sh [hunk 2]
```diff
@@ -591,15 +591,28 @@ install_from_ftp() {
...
 		DISTS="${DISTS} lib32"
 		[ -n "${KERNEL}" ] && DISTS="${DISTS} kernel"
-		fetch_file ${JAILMNT}/fromftp/MANIFEST ${URL}/MANIFEST
 		[ -s "${JAILMNT}/fromftp/MANIFEST" ] || err 1 "Empty MANIFEST file."
 		for dist in ${DISTS}; do
 			grep -q ${dist} ${JAILMNT}/fromftp/MANIFEST || continue
 			msg "Fetching ${dist} for FreeBSD ${V} ${ARCH}"
 			fetch_file ${JAILMNT}/fromftp/${dist}.txz ${URL}/${dist}.txz
+			MHASH=`awk "/^${dist}/ { print \\$2 }" ${JAILMNT}/fromftp/MANIFEST`
+			FHASH=`sha256 -q ${JAILMNT}/fromftp/${dist}`
```
Comment (bdrewery) [inline src/share/poudriere/jail.sh]: Needs .txz here. I'll add it in.

Comment (cperciva) [inline src/share/poudriere/jail.sh]: Oops, sorry about that -- I missed the `.txz` change in 4e579416 as I was updating the patch.

**Pull request merged**